scripts/typescript_checks.py

```diff
@@ -82,12 +835,91 @@ def compile_and_check_typescript(config_path: str) -> None:
     assert process.stdout is not None
     error_messages = list(iter(process.stdout.readline, ''))
 
-    if error_messages:
-        print('Errors found during compilation\n')
-        print('\n'.join(error_messages))
-        sys.exit(1)
+    if config_path == STRICT_TSCONFIG_FILEPATH:
+        # Generate file names from the error messages.
+        errors = [x.strip() for x in error_messages]
+        # Remove the empty lines and error explanation lines.
+        prefixes = ('core', 'extension', 'typings')
+        errors = [x for x in errors if x.startswith(prefixes)]
+        # Remove error explanation lines.
+        errors = [x.split('(', 1)[0] for x in errors]
+        # Remove the duplicate occurrences of the file names.
+        errors = list(dict.fromkeys(errors))
```

Comment:
The canonical way to remove duplicates is `set(errors)`

Reply:
Done